### PR TITLE
Fix torch.pow when the scalar base is a complex number

### LINKED ISSUE #43829: Poor support of torch.pow(out) for complex tensor
```python
>>> import torch
>>> r = 2. 
>>> c = 2. + 2j
>>> rt = torch.randn(5, dtype=torch.float32)
>>> ct = torch.randn(5, dtype=torch.complex64)
>>> o = torch.randn(5, dtype=torch.complex64)
>>> 
>>> torch.pow(c, rt, out=o)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: value cannot be converted to type double without overflow: (2,2)
>>> torch.pow(c, ct, out=o)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: value cannot be converted to type double without overflow: (2,2)
>>> torch.pow(rt, r, out=o)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Found dtype ComplexFloat but expected Float
```

cc @ezyang @anjali411 @dylanbespalko

### PR DESCRIPTION
**Pull request by RockingJavaBean**
Fixes #43829

### CODE REVIEW

**Comment by RockingJavaBean** [comment]
@JackCaoG This PR adds tests to check complex base numbers of the `torch.pow` function.
The xla tests failed due to precision issues.

```
Sep 25 09:34:10 FAIL [0.067s]: test_complex_scalar_pow_tensor_xla_float64 (__main__.TestTorchDeviceTypeXLA)
...
Sep 25 09:34:10 AssertionError: False is not true : Tensors failed to compare as equal! 
Imaginary parts failed to compare as equal! With rtol=0.001 and atol=0.001, 
found 100 element(s) (out of 100) whose difference(s) exceeded the margin of error (including 0 nan comparisons). 
The greatest difference was 0.5494986167779923 (0.5494986167779923 vs. 0.0), which occurred at index 19.
```

I noticed the testes for float bases are skipped in the xla CI job.
```
Sep 25 09:21:16   test_float_scalar_pow_float_tensor_xla (__main__.TestTorchDeviceTypeXLA) ... skip (0.002s)
```

Please kindly confirm whether I could annotate complex base tests with `onlyOnCPUAndCUDA` for now, and later we may add `test_complex_scalar_pow_tensor` to [pytorch_test_base.py](https://github.com/pytorch/xla/blob/1155541f5efb62150ffc2445c29035f4b146565b/test/pytorch_test_base.py#L122) to skip the tests.

**Comment by JackCaoG** [comment]
@RockingJavaBean Thanks for the heads up, I think you can mark test to be `onlyOnCPUAndCUDA ` for now.

**Review by anjali411 — APPROVED** [review]
LGTM once the complex test is skipped for xla

File: test/test_torch.py [hunk 1]
```diff
@@ -13716,6 +13716,15 @@ def test_float_scalar_pow_float_tensor(self, device):
         for base in floats:
             self._test_pow(base, tensor)
 
+    @onlyOnCPUAndCUDA
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    @dtypes(*(torch.testing.get_all_dtypes(include_bool=False, include_bfloat16=False)))
+    def test_complex_scalar_pow_tensor(self, device, dtype):
+        complexes = [0.5j, 1. + 1.j, -1.5j, 2.2 - 1.6j]
+        tensor = torch.rand(100).to(dtype=dtype, device=device)
```
Comment (zou3519) [inline test/test_torch.py:13724]: For integer dtypes this resolves to a tensor full of zeros, which may not be the most interesting test case. We have a make_tensor function to generate a random tensor that would be nicer to use: https://github.com/pytorch/pytorch/blob/ecdbea77bca044ee97eb5f17d435d701e044b79f/torch/testing/_internal/common_utils.py#L1441-L1442

Comment (RockingJavaBean) [reply test/test_torch.py:13724]: Thanks so much for pointing out this issue, and offering the kind tip of using the `make_tensor` function.
#47101 has been created to address this comment and special cases of `zero exponents` and the `1 + 0j` base are added as well, please kindly help review.

**Pull request closed without merging**